### DISCUSSION TITLE
Dark and Light mode color change

### DISCUSSION THREAD

**Discussion by jaypatel2889** [airbnb/lottie-ios#1825] [Q&A]
Hey all. Does anyone have experience working with Lottie in Dark and Light mode and changing the color in each mode in Lottie 3.5? I have tried using `ColorValueProvider`  with `AnimationKeypath` though it would require to know KeyPath  name for each code in order to map. 

I have also used `ColorValueBlock` in `ColorValueProvider` but it just returns first match if we provide dynamic filters like `**.Color`. It would nice to get all `keypath`s so colors can be mapped. 

Does anyone have worked with both mode and know how to change change colors dynamically without giving static keypath names?

**Comment by calda** [comment]
Since Lottie is primarily a series of `CALayer`s that render `CGColor`s, there isn't a great way to provide a single color that automatically varies based on the system dark mode setting. 

I think the simplest approach would probably be to have two different Lottie assets -- one for light mode, one for dark mode, and dynamically use one of the other in your app. 

Doing this through a `ColorValueProvider` requires hardcoding all of the keypath strings, like you mentioned, which is tedious and complicated for large animations. 

We don't currently have a simpler solution for this, unfortunately. Some approaches I could imagine include:
 1. Updating the `LottieColor` schema to support both a light mode color and dark mode color, within the same animation. This may not be a viable approach since most Lottie file are exported from After Effects, which I assume doesn't have this concept.
 2. Extend the [DotLottie](https://dotlottie.io/) file format to support including a light mode and dark mode variant of the animation. Then we could automatically switch between them based on the current setting, which no action needed for consumers.

**Reply by JohanForsellTibber** [reply]
Any news/progress on these ideas for dark/light mode solutions?

**Reply by mlch911** [reply]
Any news/progress on these ideas for dark/light mode solutions? (from 2024)

**Reply by nikolaiwaerpen** [reply]
Any news/progress on these ideas for dark/light mode solutions? (from 2025)

**Reply by tsaiyan** [reply]
Any news/progress on these ideas for dark/light mode solutions? (from 2026)

**Comment by ethyaan** [comment]
Since it's using paths in my case inside svg.
I used below trick to achieve my changes.
I targeted the colors that I want to change only.

```
.dark .levels svg path[stroke="rgb(18,19,49)"] {
  stroke: white;
}
```

Before

![image](https://github.com/airbnb/lottie-ios/assets/6294823/4d192380-9594-4e00-8623-900dd190393b)


After

![image](https://github.com/airbnb/lottie-ios/assets/6294823/8a4895d3-c990-435c-b015-ff43b9fc5ac3)

**Reply by mlch911** [reply]
Can you elaborate on how to do that? Your screenshot looks like it's on the web, not on iOS?

**Comment by kentslaney** [comment]
for my black and white logo, [`filter: invert(1)`](https://developer.mozilla.org/en-US/docs/Web/CSS/filter-function/invert) worked

**Reply by mlch911** [reply]
How to use this on iOS?

**Reply by kentslaney** [reply]
Oh, my mistake, this was for web like the answer above mine (probably how I got to the thread). [You can see the results](https://kentslaney.github.io/) if you're curious anyway though.
